### ISSUE TITLE
LLVM_USE_LINKER should use --ld-path= instead of -fuse-ld=

### ISSUE THREAD

**Issue by brechtsanders** [llvm/llvm-project#53565]
When building 13.0.0 with CMake variable `CMAKE_CXX_COMPILER` set to `clang` (also version 13.0.0) and `LLVM_USE_LINKER` set to `ld.lld.exe` (also version 13.0.0) the following warning shows up:
```
clang-13: warning: '-fuse-ld=' taking a path is deprecated. Use '--ld-path=' instead [-Wfuse-ld-path]
```
So somewhere in the CMake configuration something needs to be changed to avoid this warning.

The build works fine though and the error can be avoided by adding `-fuse-ld=lld` to `CMAKE_CXX_FLAGS` and not setting `LLVM_USE_LINKER`.

**Comment by keith** [comment]
I started working on this a while back https://reviews.llvm.org/D113166

**Comment by kands-code** [comment]
So, is there any progress on this issue? I've encountered this problem recently as well. These warnings, while not affecting the compilation, are quite annoying.
